src/diary.cpp

```diff
@@ -643,6 +710,7 @@ void diary::new_page()
     page -> kills = g ->get_kill_tracker().kills;
     page -> npc_kills = g->get_kill_tracker().npc_kills;
     avatar *u = &get_avatar();
+    page -> time_accuracy = u->has_watch() ? 2 : get_map().get_abs_sub().z() >= 0 ? 1 : 0;
```

Comment:
Simply being above ground doesn't necessarily gives you ability to see the sky: character could be standing inside a building with closed doors and windows. I suggest using a check for `is_outside()` instead.

Reply:
Sure, I can do that.

The "above ground" condition was just to match the way timekeeping works in the sidebar:

https://github.com/CleverRaven/Cataclysm-DDA/blob/f33592293975e6765c064c3f77a97fbccc75d7fc/src/panels.cpp#L570-L578

Reply:
```suggestion
    page -> time_accuracy = u->has_watch() ? 2 : is_creature_outside( *u ) ? 1 : 0;
```

Reply:
> The "above ground" condition was just to match the way timekeeping works in the sidebar:

This was my own work from the #20953 :confused:

While you're here, you could replace sidebar timekeeping logic too.